src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
```

Comment:
>This will depend on its scope, for example, see  storeDialogSize.

This sounds like a bad design in General. But in this case you have another error. Since we use this helper, we need to make these objects to be members of the class whose settings they handle.
>Agree. How about like this:

void set(const T &value) const
T operator() (const QVariant &defaultValue = QVariant()) const  // get

IMO, it's better to use `operator=()` and `operator T()` so we can use it as variable:
```c++
m_settingDialogSize = QSize(20, 20);
someVar = m_settingDialogSize;
```